modules/openthread/platform/infra_if.c

```diff
@@ -257,3 +277,113 @@ void infra_if_stop_icmp6_listener(void)
 	(void)net_icmp_cleanup_ctx(&rs_ctx);
 	(void)net_icmp_cleanup_ctx(&na_ctx);
 }
+
+#if defined(CONFIG_OPENTHREAD_NAT64_TRANSLATOR)
+otError infra_if_nat64_init(struct in_addr addr)
+{
+	otError error = OT_ERROR_NONE;
+	static struct net_icmp_ctx ctx;
+	struct sockaddr anyaddr = { .sa_family = AF_INET};
+
+	VerifyOrExit(net_icmp_init_ctx(&ctx, AF_INET, NET_ICMPV4_ECHO_REPLY, 0,
+				       icmpv4_handle_echo_reply) == 0,
+		     error = OT_ERROR_FAILED);
+
+	raw_udp_sock = zsock_socket(AF_INET, SOCK_RAW, IPPROTO_UDP);
+	VerifyOrExit(raw_udp_sock >= 0, error = OT_ERROR_FAILED);
+	VerifyOrExit(zsock_bind(raw_udp_sock, &anyaddr, sizeof(struct sockaddr_in)) == 0,
+		     error = OT_ERROR_FAILED);
+
+	sockfd_udp[0].fd = raw_udp_sock;
+	sockfd_udp[0].events = ZSOCK_POLLIN;
+
+	VerifyOrExit(net_socket_service_register(&handle_infra_if_udp_recv, sockfd_udp,
+						 ARRAY_SIZE(sockfd_udp), NULL) == 0,
+		     error = OT_ERROR_FAILED);
+
+exit:
+	return error;
+}
+
+static int icmpv4_handle_echo_reply(struct net_icmp_ctx *ctx,
+				    struct net_pkt *pkt,
+				    struct net_icmp_ip_hdr *hdr,
+				    struct net_icmp_hdr *icmp_hdr,
+				    void *user_data)
+{
+	otError error = OT_ERROR_NONE;
+	struct net_pkt *ot_pkt = NULL;
+
+	if (pkt != NULL) {
+		ot_pkt = net_pkt_clone(pkt, K_NO_WAIT);
+		VerifyOrExit(ot_pkt != NULL);
+		VerifyOrExit(notify_new_tx_frame(ot_pkt) == 0, error = OT_ERROR_FAILED);
+	}
+
+exit:
+	if (error != OT_ERROR_NONE) {
+		if (ot_pkt != NULL) {
+			net_pkt_unref(ot_pkt);
+		}
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static void udp_receive_handler(struct net_socket_service_event *evt)
+{
+	int len;
+	struct net_pkt *ot_pkt = NULL;
+	struct otbr_msg_ctx *req = NULL;
+	otError error = OT_ERROR_NONE;
+
+	VerifyOrExit(openthread_border_router_allocate_message((void **)&req) == OT_ERROR_NONE,
+		     error = OT_ERROR_FAILED);
+	VerifyOrExit(evt->event.revents & ZSOCK_POLLIN);
+
+	len = zsock_recv(raw_udp_sock, req->buffer, sizeof(req->buffer), 0);
+	ot_pkt = net_pkt_alloc_with_buffer(ail_iface_ptr, len, AF_INET, IPPROTO_UDP, K_NO_WAIT);
+
+	VerifyOrExit(ot_pkt != NULL, error = OT_ERROR_FAILED);
+	VerifyOrExit(net_pkt_write(ot_pkt, req->buffer, len) == 0, error = OT_ERROR_FAILED);
+
+	openthread_border_router_deallocate_message((void *)req);
+	req = NULL;
+
+	VerifyOrExit(notify_new_tx_frame(ot_pkt) == 0, error = OT_ERROR_FAILED);
+
+exit:
+	if (error != OT_ERROR_NONE) {
+		if (ot_pkt != NULL) {
+			net_pkt_unref(ot_pkt);
+		}
+		if (req != NULL) {
+			openthread_border_router_deallocate_message((void *)req);
+		}
+	}
+}
+
+otError infra_if_create_raw_socket(void)
+{
+	otError error = OT_ERROR_NONE;
+
+	raw_send_sock = zsock_socket(AF_INET, SOCK_RAW, IPPROTO_RAW);
```

Comment:
I'm a bit puzzled here, `raw_send_sock` doesn't seem to be used anywhere? And the error check below verifies `raw_udp_sock` instead (which is created elsewhere)?

Reply:
What didn't work exactly, I think it should be perfectly fine to send/receive over a single IP socket. I mean two sockets are fine too, but it'd be good to get to the bottom of this to save some resources.